src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,95 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.DukeException;
+
+/**
+ * Represents a list of tasks.
+ *
+ * @author wz2k
+ */
+public class TaskList {
+    /** Task list */
+    private ArrayList<Task> taskList;
+
+    /**
+     * Creates a list of tasks.
+     *
+     * @param taskList List of tasks.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return Total task count.
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task to add.
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+    }
+
+    /**
+     * Returns the specified task.
+     *
+     * @param taskNumber Task order in the list.
+     * @return Task of the corresponding task number.
+     */
+    public Task getTask(int taskNumber) {
+        return taskList.get(taskNumber - 1);
+    }
+
+    /**
+     * Marks the task at the specified task number as done.
+     *
+     * @param taskNumber Task order in the list.
+     * @return Task of the corresponding task number.
+     */
+    public Task markTask(int taskNumber) {
+        Task task = taskList.get(taskNumber - 1);
+        task.mark();
+        return task;
+    }
+
+    /**
+     * Marks the task at the specified task number as not done.
+     *
+     * @param taskNumber Task order in the list.
+     * @return Task of the corresponding task number.
+     */
+    public Task unmarkTask(int taskNumber) {
+        Task task = taskList.get(taskNumber - 1);
+        task.unmark();
+        return task;
+    }
+
+    /**
+     * Deletes a specified task based on its order.
+     *
+     * @param taskNumber Task order in the list.
+     * @return Task of the corresponding task number.
+     * @throws DukeException If task does not exist.
+     */
+    public Task deleteTask(int taskNumber) throws DukeException {
+        int size = taskList.size();
+
+        if (size == 0 || taskNumber > size) {
+            throw new DukeException("duke.task.Task number does not exist.");
```

Comment:
Any reason the error message is this way?

Reply:
Thanks for spotting this, might be due to intellij's auto refactoring when I was renaming